### PR TITLE
(DOCSP-31036) Adds example data federation config file.

### PR DESCRIPTION
**Pull request by erabil-mdb**
[JIRA](https://jira.mongodb.org/browse/DOCSP-31036)
[BUILD](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=64b19c258d53f9c4af26c4c9)

STAGING:
- [JSON](https://docs-mongodbcom-staging.corp.mongodb.com/atlas-cli/docsworker-xlarge/DOCSP-31036/reference/json/)
- [Atlas Data Federation Configuration File](https://docs-mongodbcom-staging.corp.mongodb.com/atlas-cli/docsworker-xlarge/DOCSP-31036/reference/json/data-federation-config-file/)

### CODE REVIEW

**Review by kanchana-mongodb — APPROVED** [review]
LGTM

File: source/reference/json/data-federation-config-file.txt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+     - Human-readable label that identifies the data store. 
+       The ``databases.[n].collections.[n].dataSources.[n].storeName`` field references 
+       this value as part of the mapping configuration. To use |service| as a data store, 
+       the {+fdi+} requires a serverless instance or an ``M10`` or higher {+cluster+}.
+
+   * - ``storage.``
+       ``stores.``
+       ``provider``
+     - string
+     - The :ref:`type of data store <config-adf>`. For example, ``atlas``, ``http``, ``online_archive``,  ``s3``, or ``DataLakeAzureBlobStore``.
```
Comment (blva) [inline source/reference/json/data-federation-config-file.txt:88]:
```suggestion
     - The :ref:`type of data store <config-adf>`. For example, ``atlas``, ``http``, ``online_archive``, ``s3``, or ``DataLakeAzureBlobStore``.
```

File: source/reference/json/data-federation-config-file.txt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+========================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+You can use an {+adf+} configuration file to specify the required 
+settings for :ref:`creating a federated database
```
Comment (blva) [inline source/reference/json/data-federation-config-file.txt:14]: when I click in this link it takes me to https://www.mongodb.com/docs/atlas/data-federation/#std-label-atlas-data-federation is this expected? looking at other json files, it usually forwards me to the command that uses the json file e.g. [backup](https://docs-mongodbcom-staging.corp.mongodb.com/atlas-cli/docsworker-xlarge/DOCSP-31036/reference/json/cloud-backup-schedule-config-file/) takes me to [command](https://www.mongodb.com/docs/atlas/cli/stable/command/atlas-backups-schedule-update/)

**Review by blva — APPROVED** [review]
1 minor question and nit

File: source/reference/json/data-federation-config-file.txt (hunk 1 above)
Comment (erabil-mdb) [reply source/reference/json/data-federation-config-file.txt]: Good catch! Fixed.

File: source/reference/json/data-federation-config-file.txt (hunk 2 above)
Comment (erabil-mdb) [reply source/reference/json/data-federation-config-file.txt:14]: Another great catch! I updated the link to point to the command.

**Pull request merged**